aspnetcore/web-api/action-return-types.md

```diff
@@ -10,6 +10,207 @@ uid: web-api/action-return-types
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultOfTProductsController.cs" id="snippet_GetByIdActionResultOfT" highlight="7":::
+
+In the preceding action:
+
+* A 404 status code is returned when the product doesn't exist in the database.
+* A 200 status code is returned with the corresponding `Product` object when the product does exist.
+
+### Asynchronous action
+
+Consider an asynchronous action in which there are two possible return types:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultOfTProductsController.cs" id="snippet_CreateAsyncActionResultOfT" highlight="9,15":::
+
+In the preceding action:
+
+* A 400 status code (<xref:Microsoft.AspNetCore.Mvc.ControllerBase.BadRequest%2A>) is returned by the ASP.NET Core runtime when:
+  * The [`[ApiController]`](xref:Microsoft.AspNetCore.Mvc.ApiControllerAttribute) attribute has been applied and model validation fails.
+  * The product description contains "XYZ Widget".
+* A 201 status code is generated by the <xref:Microsoft.AspNetCore.Mvc.ControllerBase.CreatedAtAction%2A> method when a product is created. In this code path, the `Product` object is provided in the response body. A `Location` response header containing the newly created product's URL is provided.
+
+## HttpResults type
+
+In addition to the MVC-specific built-in result types (<xref:Microsoft.AspNetCore.Mvc.IActionResult> and [ActionResult\<T>](xref:Microsoft.AspNetCore.Mvc.ActionResult%601)), ASP.NET Core includes the [HttpResults](xref:Microsoft.AspNetCore.Http.HttpResults) types that can be used in both [Minimal APIs](/aspnet/core/fundamentals/minimal-apis) and Web API.
+
+Different than the MVC-specific result types, the `HttpResults`:
+
+* Are a results implementation that is processed by a call to [IResult.ExecuteAsync](xref:Microsoft.AspNetCore.Http.IResult.ExecuteAsync%2A).
+* Does ***not*** leverage the configured [Formatters](/aspnet/core/web-api/advanced/formatting#format-specific-action-results). Not leveraging the configured formatters means:
+
+  * Some features like `Content negotiation` aren't available.
+  * The produced `Content-Type` is decided by the `HttpResults` implementation.
+
+The `HttpResults` can be useful when sharing code between Minimal APIs and Web API. 
+
+### IResult type
+
+The <xref:Microsoft.AspNetCore.Http.HttpResults> namespace contains classes that implement the <xref:Microsoft.AspNetCore.Http.IResult> interface. The `IResult` interface defines a contract that represents the result of an HTTP endpoint. The static [Results](<xref:Microsoft.AspNetCore.Http.Results>) class is used to create varying `IResult` objects that represent different types of responses.
+
+The [Built-in results](/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0&preserve-view=true#bimr7) table shows the common result helpers.
```

Comment:
Is this link expected to work already? Still not pointing to the right header.

Reply:
> Is this link expected to work already? Still not pointing to the right header.

No, you'd have to `git pull origin main` to bring it in, but when the PR merges, the link will work.